src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -151,6 +151,24 @@ public override async Task RunAsync()
             // Get the WorkerRuntime
             var workerRuntime = GlobalCoreToolsSettings.CurrentWorkerRuntime;
 
+            // Determine the appropriate default targetFramework
+            // NOTE: .NET 7.0 is only supported on dotnet-isolated
+            // TODO: Include proper steps for publishing a .NET Framework 4.8 application
+            if (workerRuntime == WorkerRuntime.dotnetIsolated)
+            {
+                string projectFilePath = ProjectHelpers.FindProjectFile(functionAppRoot);
+                if (projectFilePath != null)
+                {
+                    var projectRoot = ProjectHelpers.GetProject(projectFilePath);
+                    var targetFramework = ProjectHelpers.GetPropertyValue(projectRoot, Constants.TargetFrameworkElementName);
+                    if (targetFramework.Equals("net7.0", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
For .NET 5 Isolated project (`net5.0`), Would the `_requiredNetFrameworkVersion` value will be incorrect (6.0)? Let's make sure net5 app deployment also work.

Reply:
We don't plan on supporting .NET 5.0 deployments with Core Tools V4 since Core Tools V4 is tied to Functions V4, and according to [Supported languages in Azure Functions | Microsoft Docs](https://docs.microsoft.com/en-us/azure/azure-functions/supported-languages#languages-by-runtime-version), .NET 5.0 is supported only on Functions V3.

Reply:
Got it. Thanks!